model/host/db.go

```diff
@@ -398,6 +398,7 @@ func NeedsNewAgent(currentTime time.Time) db.Q {
 	return db.Query(bson.M{
 		StatusKey:    evergreen.HostRunning,
 		StartedByKey: evergreen.User,
+		bsonutil.GetDottedKeyName(DistroKey, distro.NoTasksAllowedKey): false,
```

Comment:
This should be NoTasksAllowed: ($ne: true} because you marked the field as omitempty